modules/light-clients/14-avalanche/update.go

```diff
@@ -0,0 +1,108 @@
+package avalanche
+
+import (
+	"fmt"
+	"reflect"
+
+	errorsmod "cosmossdk.io/errors"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
+)
+
+// VerifyClientMessage checks if the clientMessage is of type Header or Misbehaviour and verifies the message
+func (cs *ClientState) VerifyClientMessage(
+	ctx sdk.Context, cdc codec.BinaryCodec, clientStore sdk.KVStore,
+	clientMsg exported.ClientMessage,
+) error {
+	switch msg := clientMsg.(type) {
+	case *Header:
+		err := cs.verifyHeader(ctx, clientStore, cdc, msg)
+		if err != nil {
```

Comment:
Assume this was from debugging, but suggest changing this to `return cs.verifyHeader(ctx, clientStore, cdc, msg)`

Reply:
update